### DISCUSSION TITLE
API to update Patch Panel port names

### DISCUSSION THREAD

**Discussion by sfeldmann-uwh** [opendcim/openDCIM#1485] [Ideas]
I'm wondering how difficult it would be to add to the API some functionality that matches clicking on the dropdown for the patch panel patch port header and selecting custom.  I can add patch panels via the API, but would like to follow that up with another call to set the jack IDs.  In my case the pattern would be ([1st jack num];1) There are several fields throughout with similar functionality, so this could potentially cover many of them, color, media etc.

**Comment by wilpig** [comment]
not overly difficult we just haven't done much in the api with the port stuff.  it's primarily just a chore for someone to sit down and convert the class over to the api paths.  It isn't overly complicated and we'd welcome a pull for it

**Comment by alex001x** [comment]
Hi Wilpig,
can you tell me which class file is concerned and an example code, I think I will do the work on API

**Reply by wilpig** [reply]
All of the port connections are handled by https://github.com/opendcim/openDCIM/blob/master/classes/DevicePorts.class.php some of the other functions are handled by https://github.com/opendcim/openDCIM/blob/master/scripts/ajax_portnames.php and sections in the devices.php itself like https://github.com/opendcim/openDCIM/blob/master/devices.php#L101   https://github.com/opendcim/openDCIM/blob/master/devices.php#L164

**Comment by sfeldmann-uwh** [comment]
My dev group has done this build on our instance. I have talked to them about releasing it back to the project but haven’t gotten to that yet.  I also have a request with them to help automate some notes onto the patch panel ports. If you would like to take a look at what we have to verify functionality in your environment I can share it.


On Apr 22, 2024, at 7:09 AM, Wilbur Longwisch ***@***.***> wrote:

﻿
You don't often get email from ***@***.*** Learn why this is important<https://aka.ms/LearnAboutSenderIdentification>

WARNING: This email appears to have originated outside of the UW Health email system.
DO NOT CLICK on links or attachments unless you recognize the sender and know the content is safe.

All of the port connections are handled by https://github.com/opendcim/openDCIM/blob/master/classes/DevicePorts.class.php some of the other functions are handled by https://github.com/opendcim/openDCIM/blob/master/scripts/ajax_portnames.php and sections in the devices.php itself like https://github.com/opendcim/openDCIM/blob/master/devices.php#L101 https://github.com/opendcim/openDCIM/blob/master/devices.php#L164

**Reply by alex001x** [reply]
yes, thks

**Comment by sfeldmann-uwh** [comment]
This has been uploaded here: https://github.com/opendcim/openDCIM/pull/1535


Scott Feldmann
Infrastructure Principal
UW Health - Information Services
***@***.***
